river/Seat.zig

```diff
@@ -164,9 +165,21 @@ pub fn focus(self: *Self, _target: ?*View) void {
             self.focus_stack.push(new_focus_node);
         }
 
+        if (self.input_manager.pointer_constraints.constraintForSurface(view.surface.?, self.wlr_seat)) |constraint| {
+            @intToPtr(*PointerConstraint, constraint.data).setAsActive();
+        } else if (self.cursor.constraint != null) {
+            PointerConstraint.warpToHint(&self.cursor);
+            self.cursor.constraint.?.sendDeactivated();
+            self.cursor.constraint = null;
+        }
```

Comment:
This seems strange, why would we want to deactivate the constraint when a surface gains focus?

Reply:
I don't think we want the cursor to be constrained to a surface that we are not focusing.
i.e. when switching focus from a game to a web browser.